### PR TITLE
Hotfix/4.3.3

### PR DESCRIPTION
**Pull request by ceo1647**
restyle Filter builder. Add apply button and made preview optional.
fb2

### CODE REVIEW

**Comment by GeertvanHorrik** [comment]
I thikn the apply button should be part of the window buttons?

**Comment by ceo1647** [comment]
another restyle without Apply button

**Review by Averus-a — APPROVED** [review]

File: src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+            var isEnabled = (bool)values[0];
+            if (isEnabled)
+            {
+                var isPreviewVisible = (bool)values[1];
+
+                return $"{(isPreviewVisible ? "Hide" : "Show")} {values[2]} items of {values[3]}";
+            }
+            
+            return "Not filtered";
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs:29]: move to resources

File: src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            {
+                return string.Empty;
+            }
+
+            var isEnabled = (bool)values[0];
+            if (isEnabled)
+            {
+                var isPreviewVisible = (bool)values[1];
+
+                return $"{(isPreviewVisible ? "Hide" : "Show")} {values[2]} items of {values[3]}";
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs:26]: move to resources

File: src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs [hunk 3]
```diff
@@ -71,6 +63,7 @@ public class EditFilterViewModel : ViewModelBase
             AddGroupCommand = new Command<ConditionGroup>(OnAddGroup);
             AddExpressionCommand = new Command<ConditionGroup>(OnAddExpression);
             DeleteConditionItem = new Command<ConditionTreeItem>(OnDeleteCondition, OnDeleteConditionCanExecute);
+            ShowHidePreview = new Command(OnShowHidePreview);
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs:66]: TogglePreview

File: src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs [hunk 4]
```diff
@@ -89,15 +82,18 @@ public override string Title
         public bool EnableAutoCompletion { get; private set; }
         public bool AllowLivePreview { get; private set; }
         public bool EnableLivePreview { get; set; }
-
+        public bool IsLivePreviewDirty { get; set; }
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs:85]: public setter really required?

File: src/Orc.FilterBuilder.Xaml/Views/EditFilterView.xaml [hunk 5]
```diff
@@ -307,35 +252,64 @@
...
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="Auto"/>
+                <ColumnDefinition Width="Auto"/>
+                <ColumnDefinition Width="*"/>
+            </Grid.ColumnDefinitions>
+
+            <CheckBox x:Name ="LivePreviewCheckBox"
+                      HorizontalAlignment="Left"
+                      FontWeight="SemiBold"
+                      Content="Live update"
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder.Xaml/Views/EditFilterView.xaml:268]: LanguageBinding

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]

File: src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs (hunk 3 above)
Comment (ceo1647) [reply src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs:66]: done

File: src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs (hunk 4 above)
Comment (ceo1647) [reply src/Orc.FilterBuilder.Xaml/ViewModels/EditFilterViewModel.cs:85]: should be removed...also made other refactoring in property accessors

File: src/Orc.FilterBuilder.Xaml/Views/EditFilterView.xaml (hunk 5 above)
Comment (ceo1647) [reply src/Orc.FilterBuilder.Xaml/Views/EditFilterView.xaml:268]: done

File: src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs (hunk 2 above)
Comment (ceo1647) [reply src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs:26]: done

File: src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs (hunk 1 above)
Comment (ceo1647) [reply src/Orc.FilterBuilder.Xaml/Converters/FilterResultMultiValueConverter.cs:29]: done

**Pull request closed without merging**